### PR TITLE
Fixed bug with ordering by a concatenated field in DoctrineListBuilder

### PR DESCRIPTION
**Pull request by kriswillis**
Adding a `DoctrineConcatenationFieldDescriptor` to a `DoctrineListBuilder` and then trying to order by in within an admin iterface throws the following exception:
> QueryException: [Syntax Error] line 0, col 597: Error: Expected end of string, got '('

It seems you can only order by *fields* that have been selected in the query.

__Tasks__

- [x] Test coverage
- [x] Update changelog

__Information__

| Q                | A
| ---------------- | ---
| Fixed tickets    | N/A
| BC breaks        | None
| Documentation PR | N/A

### CODE REVIEW

File: src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php [hunk 1]
```diff
@@ -183,7 +184,11 @@ function ($array) {
     protected function assignSortFields($queryBuilder)
     {
         foreach ($this->sortFields as $index => $sortField) {
-            $queryBuilder->addOrderBy($sortField->getSelect(), $this->sortOrders[$index]);
+            $sort = ($sortField instanceof DoctrineConcatenationFieldDescriptor)
+                ? $sortField->getName()
+                : $sortField->getSelect();
```
Comment (danrot) [inline src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php]: Couldn't this ternary operator be avoided if we just always use `$sortField->getName()`? Then we could also get rid of this additional variable.

Comment (kriswillis) [inline src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php]: @danrot That's how I approached this originally but it appeared to break a bunch of other bundles (https://travis-ci.org/sulu-io/sulu/jobs/77311121), I didn't have the time to investigate it further at that point so compromised with the ternary.

**Comment by chirimoya** [comment]
@kriswillis @danrot ping

File: src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php (hunk 1 above)
Comment (danrot) [inline src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php]: Can you maybe check the exact error? Just seeing that the status code is 500 is not really enough to argue about that :-)

**Comment by danrot** [comment]
This PR is replaced by #2933, because only the test added here is missing. The actual bug seems to be fixed already, as you can [see in the code](https://github.com/sulu/sulu/blob/0a9a98f49106286f58d0e32562918922c3da74f7/src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php#L235)

**Pull request closed without merging**